ruby/bindings/InitRubyBindings.cpp

```diff
@@ -986,27 +988,67 @@ ENV.reject! { |k, _| ['GEM', 'BUNDLE'].any? { |x| k.start_with?(x) } }
         end
       end
 
-
       if locked_specs.none? { |locked_spec| locked_spec.dependencies.any? {|x| x.name == spec.name} }
-        locked_spec_v = locked_specs.select{|locked_spec| locked_spec.name == spec.name}.first.version
-        $logger.debug("Removing native gem #{spec.name} from Bundler locked_specs (version #{locked_spec_v}), using the embedded dependency (CLI one has version #{spec.version})")
+        locked_spec = locked_specs.select{|locked_spec| locked_spec.name == spec.name}.first
+        $logger.debug("Removing native gem #{spec.name} from Bundler locked_specs (version #{locked_spec.version}), using the embedded dependency (CLI one has version #{spec.version})")
         $logger.trace("Gems having #{spec.name} as a dependency: #{dep_to_requirements}")
+        locked_spec.dependencies.each do |dep|
+          $logger.trace("Activating dependency #{dep.name} first")
+          found_spec = embedded_gems_to_activate.find { |x| x.name == dep.name }
+          raise "Can't find #{dep.name}" unless found_spec
+
+          next if found_spec.activated
+
+          # found_spec.load_paths.reject {|path| $LOAD_PATH.include?(path) }
+          Gem::Specification.add_spec(found_spec)  # Will add it to Gem::Specification.stubs too
+          raise "Failed to activate #{spec.name}" unless found_spec.activate
+          Bundler.rubygems.mark_loaded(found_spec)
+          stub = Gem::Specification.stubs_for(found_spec.name).first
+          bundler_stub = Bundler::StubSpecification.from_stub(stub)
+          if !locked_specs[dep].nil?
+            x = locked_specs[dep].first
+            x.instance_variable_set(:@version, found_spec.version)
+            class <<x
+              private
+              def use_exact_resolved_specifications?
+                @use_exact_resolved_specifications = false
+              end
+            end
+            #  locked_specs.delete_by_name(dep.name)
+            # locked_specs[dep] = bundler_stub
+            extra_load_paths += found_spec.full_require_paths
+          end
+        end
+
+        Gem::Specification.add_spec(spec)
         raise "Failed to activate #{spec.name}" unless spec.activate
+        Bundler.rubygems.mark_loaded(spec)
+        stub = Gem::Specification.stubs_for(spec.name).first
+        bundler_stub = Bundler::StubSpecification.from_stub(stub)
+        raise "That shouldn't happen" if locked_specs[spec.name].nil?
+        # locked_specs.delete_by_name(spec.name)
+        # locked_specs[spec.name] = bundler_stub
+
+        extra_load_paths += spec.full_require_paths
```

Comment:
Pfff. Seriously. That needs cleaning up  I guess, I'm not really using the stubs: I thought I could swap the Bundler::LazySpecification with the Bundler::StubSpecification instead, but that doesn't work.

Anyways, the idea is to try to activate the dependencies of the spec before the spec is activated...

And I'm storing the load paths that are added